tests/topotests/zebra_rib/test_zebra_rib.py

```diff
@@ -76,6 +76,34 @@ def teardown_module(mod):
     tgen = get_topogen()
     tgen.stop_topology()
 
+def test_zebra_kernel_route_vrf():
+    "Test kernel routes should be removed after interface changes vrf"
+    logger.info("Test kernel routes should be removed after interface changes vrf")
+    vrf="RED"
+    tgen = get_topogen()
+    r1 = tgen.gears["r1"]
+
+    # Add kernel routes, the interface is initially in default vrf
+    r1.run("ip route add 3.5.1.0/24 via 192.168.210.1 dev r1-eth0")
+    json_file = "{}/r1/v4_route_1_vrf_before.json".format(CWD)
+    expected = json.loads(open(json_file).read())
+    test_func = partial(
+            topotest.router_json_cmp, r1, "show ip route 3.5.1.0/24 json", expected)
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
+    assert result is None, '"r1" JSON output mismatches'
+
+    # Change the interface's vrf
+    r1.run("ip link add {} type vrf table 1".format(vrf))
+    r1.run("ip link set {} up".format(vrf))
+    r1.run("ip link set dev r1-eth0 master {}".format(vrf))
+    test_func = partial(
+            topotest.router_json_cmp, r1, "show ip route 3.5.1.0/24 json", expected)
+    _, result = topotest.run_and_expect(test_func, None, count=5, wait=1)
```

Comment:
Here should be `not None` I think, otherwise we don't wait 5 seconds and assert immediately.

Reply:
I have always tested this test before pushed, it should fail with old code and succeed with new code fixed by this PR.

You mentioned two issues:
1) Yes, it should be `(count=1, wait=5)` for that.
2) No, it really means `result`. After interface is moved, `show ip route <Net> json` should be different with origin. So, `result` should be not None.

Anyway, it is not good. So i re-modified it to be clear, and re-pushed, thanks.